aea/contracts/base.py

```diff
@@ -202,6 +202,79 @@ def get_state(
         """
         raise NotImplementedError
 
+    @classmethod
+    def contract_method_call(
+        cls, ledger_api: LedgerApi, method_name: str, **kwargs: Any
+    ) -> Optional[JSONLike]:
+        """
+        Make a contract call.
+
+        :param ledger_api: the ledger apis.
+        :param method_name: the contract method name.
+        :param kwargs: keyword arguments.
+        :return: the call result
+        """
+
+        contract_interface = cls.contract_interface.get(ledger_api.identifier, {})
+        result = ledger_api.contract_method_call(
+            contract_interface, method_name, **kwargs
+        )
```

Comment:
I think this could be done better like this:
```suggestion
        contract_instance = cls.get_instance(ledger_api, contract_address)
        result = ledger_api.contract_method_call(
            contract_instance, method_name, **kwargs
        )
```
Then you'd reuse existing logic more rather than duplicate it.

Reply:
Also note, you need the contract address not in all cases, so this can be an optional arg

Reply:
But either works - you can leave as is if it suits you more.